### ISSUE TITLE
Can spreadsheet registration go "stale"?

### ISSUE THREAD

**Issue by jennybc** [jennybc/gspreadr#15]
@jzhaoo Just FYI. Nothing you can do about this right now.

In the version I'm working on (still not pushed), I've noticed that if I register a spreadsheet (new vocabulary for what used to be "open"), then use that object several hours later to, say, access the list or cells feed for a worksheet within that spreadsheet, I get this error:

```sh
 Error in gsheets_GET(this_ws$listfeed) : 
  Was expecting content-type to be:
application/atom+xml; charset=UTF-8
but instead it's:
application/binary
```

Note I've redefined `gsheets_GET()`. But the important thing is I don't think we're properly catching and interpreting this error message in the `GET`.

If I re-register the spreadsheet, it triggers "Auto-refreshing stale OAuth token.". Then the exact same list- or cells feed request works fine. I suspect we need to improve `gsheets_GET()`'s handling of errors. Or just get a better understanding of how registered spreadsheets get "stale".

**Comment by jennybc** [comment]
I think there is more to our use of `httr` than meets my eye … I don't know, cookies are something? The time gap between registration and further queries clearly matters and re-registering, triggering a refresh of the OAuth token, does the trick.

**Comment by jennybc** [comment]
I don't want to stop and do this now but first experiment should be to use old, potentially "stale" spreadsheet object and, before use, trigger OAuth token refresh. I suspect that is the only issue. That is, I suspect nothing in the object where we store a registered spreadsheet is perishable.

BTW do we have a way to trigger OAuth token refresh explicitly? Maybe we could also make this happen automagically in the above scenario.

**Comment by jennybc** [comment]
No, the spreadsheet registration does not go stale. The OAuth token does and we need to extend auto-refresh to happen in more situations.

`register()` triggers auto-refresh when necessary.

But consuming data via, e.g., `get_via_lf()` does not, even when it should. We just get the wrong stuff back from the `GET()` and pass a mysterious message back to the user about getting content-type application/binary when we were expecting application/atom+xml.

Closing because this has merged with #20.
